src/UI/IdentityBuilderUIExtensions.cs

```diff
@@ -33,8 +33,13 @@ public static IdentityBuilder AddDefaultUI(this IdentityBuilder builder)
         {
             AddAdditionalApplicationParts(builder);
 
-            builder.Services.ConfigureOptions<IdentityDefaultUIConfigureOptions>();
+            builder.Services.ConfigureOptions(
+                typeof(IdentityDefaultUIConfigureOptions<>)
```

Comment:
There is. I close on the user type here and I can just rely on it in the configuration class. This is the pattern Identity follows for services that depend on the user.

Reply:
I'm with @pranavkm seems easier to just pass in UserType and that lets you get rid some of the TUser : IdentityUser constraints as well

Reply:
@HaoK @pranavkm That will not work. I need to pass in the type (I can't create an instance) because IdentityDefaultUIConfigureOptions has dependencies from DI (IHostingEnvironment).

Passing the user type is more involved/impossible as it would require to pass in a factory method and resolve within that method the hosting environment and pass in the user type.

Reply:
So my concern mostly is that this looks a bit like generic hell where we have a whole lot of complexity just trying to tunnel the TUser to the right place...  This now going to show up in the scaffolded code too?

Reply:
Basically I think the whole public non generic model that throws not implemented + internal derived generic model is kinda gross

Reply:
No

Reply:
Okay, if this weird model stuff isn't going to show up in the scaffolded code, its not as much of an issue